framework/include/userobjects/ElementSubdomainModifier.h

```diff
@@ -78,15 +78,25 @@ class ElementSubdomainModifier : public ElementUserObject
   // Set the name of the complement moving boundary. Create the nodeset/sideset if not exist.
   void setComplementMovingBoundaryName(MooseMesh & mesh);
 
-  // Update the moving and complement moving boundaries (both the underlying sideset and nodeset)
-  void updateBoundaryInfo(MooseMesh & mesh, const std::vector<const Elem *> & moved_elems);
+  // Update the moving boundary (both the underlying sideset and nodeset)
+  void updateMovingBoundaryInfo(MooseMesh & mesh, const std::vector<const Elem *> & moved_elems);
+
+  // Update the complement boundary (both the underlying sideset and nodeset)
+  void updateComplementBoundaryInfo(MooseMesh & mesh,
+                                    const std::vector<const Elem *> & moved_elems);
```

Comment:
Just as an FYI, we should get rid of the complement boundary functionality IMO. I have a separate UPO that updates the complement boundary, which simplifies this object quite a bit. I should clean that up and make a PR.

Reply:
Oh that would be awesome! This UO is becoming more and more convoluted and harder to navigate -- I had the concern that separating the routines for the two boundaries would not be welcomed so I am really glad you already are planning&working on this. 

Would it work to keep this routine as is in this PR and remove later in your PR? What do you suggest?